src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/cargo/CargoUtils.java

```diff
@@ -139,48 +141,50 @@ public static ItemStack insert(Block node, Block target, ItemStack stack, int in
         if (!matchesFilter(node, stack, index)) return stack;
 
         DirtyChestMenu menu = getChestMenu(target);
+        if (menu == null) {
+            if (BlockUtils.hasInventory(target)) {
+                BlockState state = target.getState();
+                if (state instanceof InventoryHolder) {
+                    return insertIntoVanillaInventory(stack, ((InventoryHolder) state).getInventory());
+                }
+            }
+            return stack;
+        }
 
-        if (menu != null) {
-            ItemStackWrapper wrapper = new ItemStackWrapper(stack);
-
-            for (int slot : menu.getPreset().getSlotsAccessedByItemTransport(menu, ItemTransportFlow.INSERT, stack)) {
-                ItemStack is = menu.getItemInSlot(slot) == null ? null : menu.getItemInSlot(slot).clone();
+        ItemStackWrapper wrapper = new ItemStackWrapper(stack);
 
-                if (is == null) {
-                    menu.replaceExistingItem(slot, stack.clone());
-                    return null;
-                }
-                else if (SlimefunUtils.isItemSimilar(new CustomItem(is, 1), wrapper, true) && is.getAmount() < is.getType().getMaxStackSize()) {
-                    int amount = is.getAmount() + stack.getAmount();
+        for (int slot : menu.getPreset().getSlotsAccessedByItemTransport(menu, ItemTransportFlow.INSERT, stack)) {
+            ItemStack itemInSlot = menu.getItemInSlot(slot);
+            if (itemInSlot == null) {
+                menu.replaceExistingItem(slot, stack);
+                return null;
+            }
 
-                    if (amount > is.getType().getMaxStackSize()) {
-                        is.setAmount(is.getType().getMaxStackSize());
-                        stack.setAmount(amount - is.getType().getMaxStackSize());
-                    }
-                    else {
-                        is.setAmount(amount);
-                        stack = null;
-                    }
+            int inSlotMaxSize = itemInSlot.getType().getMaxStackSize();
+            int inSlotAmount = itemInSlot.getAmount();
```

Comment:
currentAmount  would probably be a more sensible name for this variable here. "inSlotAmount" is rather unpleasant to read.